tests/test_models/test_inference/test_predict.py

```diff
@@ -276,16 +277,122 @@ def test_predict_in_sample_suffix(self, model, transforms, example_tsds):
     @pytest.mark.parametrize(
         "model, transforms",
-        [],
+        [
+            (CatBoostPerSegmentModel(), [LagTransform(in_column="target", lags=[2, 3])]),
+            (CatBoostMultiSegmentModel(), [LagTransform(in_column="target", lags=[2, 3])]),
+            (LinearPerSegmentModel(), [LagTransform(in_column="target", lags=[2, 3])]),
+            (LinearMultiSegmentModel(), [LagTransform(in_column="target", lags=[2, 3])]),
+            (ElasticPerSegmentModel(), [LagTransform(in_column="target", lags=[2, 3])]),
+            (ElasticMultiSegmentModel(), [LagTransform(in_column="target", lags=[2, 3])]),
+            (SARIMAXModel(), []),
+            (AutoARIMAModel(), []),
+            (MovingAverageModel(window=3), []),
+            (NaiveModel(lag=3), []),
+            (SeasonalMovingAverageModel(), []),
+        ],
     )
-    def test_predict_in_sample_suffix_failed_not_implemented_in_sample(self, model, transforms, example_tsds):
-        _test_prediction_in_sample_suffix(example_tsds, model, transforms, method_name="predict", num_skip_points=50)
+    def test_predict_in_sample_suffix_int_timestamp(self, model, transforms, example_tsds):
+        ts_int_timestamp = convert_ts_to_int_timestamp(example_tsds, shift=10)
+        _test_prediction_in_sample_suffix(
+            ts_int_timestamp, model, transforms, method_name="predict", num_skip_points=50
+        )
+
+    @to_be_fixed(raises=Exception)
+    @pytest.mark.parametrize(
+        "model, transforms",
+        [
+            (ProphetModel(), []),
+            (HoltModel(), []),
+            (HoltWintersModel(), []),
+            (SimpleExpSmoothingModel(), []),
+            (DeadlineMovingAverageModel(window=1), []),
+            (BATSModel(use_trend=True), []),
+            (TBATSModel(use_trend=True), []),
+            (StatsForecastARIMAModel(), []),
+            (StatsForecastAutoARIMAModel(), []),
+            (StatsForecastAutoCESModel(), []),
+            (StatsForecastAutoETSModel(), []),
+            (StatsForecastAutoThetaModel(), []),
+            (PatchTSModel(encoder_length=7, decoder_length=7, trainer_params=dict(max_epochs=1)), []),
+            (
+                DeepStateModel(
+                    ssm=CompositeSSM(seasonal_ssms=[WeeklySeasonalitySSM()]),
+                    input_size=1,
+                    encoder_length=7,
+                    decoder_length=7,
+                    trainer_params=dict(max_epochs=1),
+                ),
+                [SegmentEncoderTransform()],
+            ),
+        ],
+    )
+    def test_predict_in_sample_suffix_int_timestamp_failed(self, model, transforms, example_tsds):
+        ts_int_timestamp = convert_ts_to_int_timestamp(example_tsds, shift=10)
+        _test_prediction_in_sample_suffix(
+            ts_int_timestamp, model, transforms, method_name="predict", num_skip_points=50
+        )
+
+    @to_be_fixed(raises=NotImplementedError, match="Method predict isn't currently implemented")
```

Comment:
Have we not implemented it yet?  
what a shame...

Reply:
Yes, it still isn't implemented. Let's at least create an issue about it.